demo/site/src/common/components/Button.tsx

```diff
@@ -8,7 +8,7 @@ export type ButtonVariant = "contained" | "outlined" | "text";
 type ButtonProps<T extends ElementType> = {
     as?: T;
     variant?: ButtonVariant;
-} & ComponentProps<T>;
+} & Omit<ComponentProps<T>, "variant">;
```

Comment:
This is a good solution. 👍🏼  Should we exclude the other props (as, etc.) as well?

Reply:
@SebiVPS what do you think? In Typography, props such as `bottomSpacing` are typed as any if we don't omit them.